test/test_torch.py

```diff
@@ -6947,15 +6947,89 @@ def compare_strides(s1, s2, div):
 
     @skipMeta
     @dtypes(*torch.testing.get_all_dtypes())
-    def test_dlpack_conversion(self, device, dtype):
+    def test_dlpack_capsule_conversion(self, device, dtype):
         # DLpack does not explicitly support bool
         # It does it through uint8 type
         if dtype is torch.bool:
             return
+        if 'xla' in device:
+            return
         x = make_tensor((5,), device, dtype, low=-9, high=9)
         z = from_dlpack(to_dlpack(x))
         self.assertEqual(z, x)
 
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_protocol_conversion(self, device, dtype):
+        # DLpack does not explicitly support bool
+        # It does it through uint8 type
+        if dtype is torch.bool:
+            return
+        if 'xla' in device:
+            return
+        x = make_tensor((5,), device, dtype, low=-9, high=9)
+        z = from_dlpack(x)
+        self.assertEqual(z, x)
+
+    @skipMeta
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_dlpack_conversion_with_streams(self, device, dtype):
+        # DLpack does not explicitly support bool
+        # It does it through uint8 type
+        if dtype is torch.bool:
+            return
+        if 'xla' in device:
+            return
+        # Create a stream where the tensor will reside
+        if device == 'cuda':
+            x = make_tensor((5,), device, dtype, low=-9, high=9)
+            stream = torch.cuda.Stream()
+            with torch.cuda.stream(stream):
+                assert stream.query()
+                z = from_dlpack(x)
+                assert not stream.query()
+            assert stream.query()
+            self.assertEqual(z, x)
```

Comment:
Question: is `z` guaranteed to be ready when we go out of the scope of a PyTorch stream? If not, then this assert should be moved to within the scope.

Reply:
when going out of the scope of a stream, I think that all the work queued on it should be done, but I am not sure now.